### PR TITLE
Add a nightly run of the test suite against chia-blockchain at main

### PR DESCRIPTION
**Pull request by hoffmang9**

### CODE REVIEW

**Comment by hoffmang9** [comment]
And another bug uncovered - this time it's in the in memory db test change - https://github.com/Chia-Network/chia-blockchain/pull/16637:
```
ImportError while loading conftest '/home/runner/work/CAT-admin-tool/CAT-admin-tool/tests/conftest.py'.
tests/conftest.py:6: in <module>
    from chia.simulator.setup_nodes import setup_simulators_and_wallets_service
venv/lib/python3.10/site-packages/chia/simulator/setup_nodes.py:24: in <module>
    from chia.simulator.setup_services import (
venv/lib/python3.10/site-packages/chia/simulator/setup_services.py:57: in <module>
    from tests.util.db_connection import generate_in_memory_db_uri
E   ModuleNotFoundError: No module named 'tests'
Error: Process completed with exit code 4.
```

**Comment by AmineKhaldi** [comment]
> And another bug uncovered - this time it's in the in memory db test change - [Chia-Network/chia-blockchain#16637](https://github.com/Chia-Network/chia-blockchain/pull/16637):
> 
> ```
> ImportError while loading conftest '/home/runner/work/CAT-admin-tool/CAT-admin-tool/tests/conftest.py'.
> tests/conftest.py:6: in <module>
>     from chia.simulator.setup_nodes import setup_simulators_and_wallets_service
> venv/lib/python3.10/site-packages/chia/simulator/setup_nodes.py:24: in <module>
>     from chia.simulator.setup_services import (
> venv/lib/python3.10/site-packages/chia/simulator/setup_services.py:57: in <module>
>     from tests.util.db_connection import generate_in_memory_db_uri
> E   ModuleNotFoundError: No module named 'tests'
> Error: Process completed with exit code 4.
> ```

I sent an addendum [PR #16752](https://github.com/Chia-Network/chia-blockchain/pull/16752), shortly after the [original PR #16637](https://github.com/Chia-Network/chia-blockchain/pull/16637), that should address this.

File: .github/workflows/nightly-against-main.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  cancel-in-progress: true
+
+jobs:
+  build:
+    name: All tests
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      max-parallel: 4
```
Comment (cmmarslender) [inline .github/workflows/nightly-against-main.yml]: Unless doing a matrix, this doesn't really do much

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategymax-parallel

**Review by cmmarslender — CHANGES_REQUESTED** [review]

**Review by cmmarslender — APPROVED** [review]

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**